prow/cmd/pipeline/controller.go

```diff
@@ -346,14 +347,17 @@ func (c *controller) now() metav1.Time {
 	return metav1.Now()
 }
 
-func (c *controller) pipelineID(pj prowjobv1.ProwJob) (string, string, error) {
+func (c *controller) pipelineID(pj prowjobv1.ProwJob) (string, error) {
 	id, err := pjutil.GetBuildID(pj.Spec.Job, c.totURL)
 	if err != nil {
-		return "", "", err
+		return "", err
 	}
 	pj.Status.BuildID = id
-	url := pjutil.JobURL(c.config().Plank, pj, logrus.NewEntry(logrus.StandardLogger()))
-	return id, url, nil
+	return id, nil
+}
+
+func (c *controller) jobURL(pj *prowjobv1.ProwJob) string {
```

Comment:
Given the usage `pj prowjobv1.ProwJob` seems like a better signature that doesn't risk a null pointer exception